src/runtime_src/tools/scripts/_xbreset.py

```diff
@@ -0,0 +1,148 @@
+        formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument("-d","--dbdf", required=True,
+        help="DBDF of a FPGA node, with format xxxx:xx:xx.x")
+    parser.add_argument("-y","--yes", help="automatic yes to prompts",
+        action="store_true")
+    args = parser.parse_args()
+
+    get_node_bus_mapping()
+    mgmt = args.dbdf[:len(args.dbdf)-1] + "0"
+    user = args.dbdf[:len(args.dbdf)-1] + "1"
+
+    try:
+        if mgmt not in node_bus_mapping:
+            print("ERROR: Is %s a FPGA node?" % user)
+            sys.exit(1)
+        with open(os.path.join(rootDir, user, "ready")) as f : ready = f.read()
+        ready = ready.strip("\n")
+        if ready == "0x0" or ready == "0":
+            print("ERROR: FPGA %s is not usable" % user)
+            sys.exit(1)
+
+        #print node_parent_mapping
+        nodes_in_cards = bus_children[node_bus_mapping[mgmt]]
+        #print nodes_in_cards
+        if len(nodes_in_cards) == 1:
+            #print("xbutil reset %s" % user)
+            print("All existing processes will be killed.")
+            while not args.yes:
+                line = input("Are you sure you wish to proceed? [y/n]:")
+                if line == "n":
+                    sys.exit(1)
+                if line == "y":
+                    break
+            run_reset(["/opt/xilinx/xrt/bin/unwrapped/xbutil", "reset", "-d", user])
+        elif (len(nodes_in_cards) == 2):
+            if mgmt == nodes_in_cards[0]:
+                rm = nodes_in_cards[1]
+            else:
+                rm = nodes_in_cards[0]
+            print("Card level reset. This will reset all FPGAs on the card: %s, %s" % (mgmt, rm))
+            print("All existing processes will be killed.")
+            while not args.yes:
+                line = input("Are you sure you wish to proceed? [y/n]:")
+                if line == "n":
+                    sys.exit(1)
+                if line == "y":
+                    break
+            rm_user = rm[:len(rm)-1] + "1"
+            print("shutdown: %s" % rm_user)
+            run("echo 2 > " + os.path.join(rootDir, rm_user, "shutdown"))
+            print("remove: %s, %s" % (rm_user, rm))
+            run("echo 1 > " + os.path.join(rootDir, node_parent_mapping[rm], "remove"))
+            print("xbutil reset %s" % user)
+            run_reset(["/opt/xilinx/xrt/bin/unwrapped/xbutil", "reset", "-a", "-d", user])
+            print("rescan pci")
+            run("echo 1 > /sys/bus/pci/rescan")
+    except (subprocess.TimeoutExpired):
+        #xbutil reset timeout, try remove it
+        print("shutdown: %s" % user)
+        run("echo 2 > " + os.path.join(rootDir, user, "shutdown"))
+        print("remove: %s, %s" % (user, mgmt))
+        run("echo 1 > " + os.path.join(rootDir, node_parent_mapping[mgmt], "remove"))
+        print("rescan pci")
+        run("echo 1 > /sys/bus/pci/rescan")
+    except Exception as e:
+        print(e)
+
+main()
```

Comment:
This is the safe way of running the code:
```
if __name__ == "__main__":
    main()
```

Reply:
will change